open-banking-accelerator/internal-apis/internal-webapps/com.wso2.openbanking.accelerator.dcr.endpoint/src/main/java/com/wso2/openbanking/accelerator/identity/dcr/endpoint/impl/api/ClientRegistrationApiImpl.java

```diff
@@ -173,19 +173,14 @@ public Response registerClientIdPut(@Context HttpServletRequest request, @Contex
             Map<String, Object> requestAttributes = (Map<String, Object>)
                     gson.fromJson(registrationRequestDetails, Map.class);
 
-            //decode SSA
-            if (registrationRequest.getSoftwareStatement() == null) {
-                return Response.status(Response.Status.BAD_REQUEST).entity(RegistrationUtils
-                                .getErrorDTO(DCRCommonConstants.INVALID_META_DATA,
-                                        "Required parameter software statement cannot be null"))
-                        .build();
-            }
-            String ssaBody = JWTUtils.decodeRequestJWT(registrationRequest.getSoftwareStatement(), "body")
-                    .toString();
-            Map<String, Object> ssaAttributesMap = gson.fromJson(ssaBody, Map.class);
-            //RegistrationRequest registrationRequest = RegistrationUtils.getRegistrationRequest(requestAttributeMap);
             registrationRequest.setRequestParameters(requestAttributes);
-            registrationRequest.setSsaParameters(ssaAttributesMap);
+
+            if (registrationRequest.getSoftwareStatement() != null) {
+                String ssaBody = JWTUtils.decodeRequestJWT(registrationRequest.getSoftwareStatement(), "body")
```

Comment:
Shall we use constants to strings like "body: and "id" since they are used in multiple places

Reply:
Resolved in [88d0ac2](https://github.com/wso2/financial-open-banking/pull/62/commits/88d0ac2834e0ceba81ceeca4486dd9ac52acd3cc)